src/main/java/org/elasticsearch/search/aggregations/reducers/derivative/DerivativeReducer.java

```diff
@@ -88,51 +91,74 @@ public InternalAggregation reduce(InternalAggregation aggregation, ReduceContext
         InternalHistogram.Factory<? extends InternalHistogram.Bucket> factory = histo.getFactory();
 
         List newBuckets = new ArrayList<>();
+        Long lastBucketKey = null;
         Double lastBucketValue = null;
         for (InternalHistogram.Bucket bucket : buckets) {
+            Long thisBucketKey = resolveBucketKeyAsLong(bucket);
             Double thisBucketValue = resolveBucketValue(histo, bucket, bucketsPaths()[0], gapPolicy);
             if (lastBucketValue != null) {
-                double diff = thisBucketValue - lastBucketValue;
-
-                List<InternalAggregation> aggs = new ArrayList<>(Lists.transform(bucket.getAggregations().asList(), AGGREGATION_TRANFORM_FUNCTION));
-                aggs.add(new InternalSimpleValue(name(), diff, formatter, new ArrayList<Reducer>(), metaData()));
+                double gradient = thisBucketValue - lastBucketValue;
+                double xDiff = -1;
+                if (xAxisUnits != -1) {
```

Comment:
Even though this one should work, exact double comparisons tend to scare me a bit: should we use `null` instead of `-1` for non-existing x-axis units? (and store it in a Double)